tests/std/tests/P0896R4_ranges_alg_permutations/test.cpp

```diff
@@ -0,0 +1,264 @@
+                assert(equal(input, Expected[i]));
+            }
+
+            R range{input};
+            const same_as<next_permutation_result<iterator_t<R>>> auto result =
+                next_permutation(range.begin(), range.end(), ranges::less{}, get_val);
+            assert(result.in == range.end());
+            assert(!result.found);
+            assert(equal(input, Expected[0]));
+        }
+    }
+};
+
+template <auto& Expected>
+struct prev_perm_instantiator {
+    template <ranges::bidirectional_range R>
+    static constexpr void call() {
+        using ranges::prev_permutation, ranges::prev_permutation_result, ranges::equal, ranges::iterator_t;
+
+        constexpr auto number_of_permutations = static_cast<int>(ranges::size(Expected));
+
+        { // Validate range overload
+            int_wrapper input[ranges::size(Expected[0])];
+            ranges::copy(Expected[number_of_permutations - 1], input);
+
+            for (int i = number_of_permutations - 1; i-- > 0;) {
+                R range{input};
+                const same_as<prev_permutation_result<iterator_t<R>>> auto result =
+                    prev_permutation(range, ranges::less{}, get_val);
+                assert(result.in == range.end());
+                assert(result.found);
+                assert(equal(input, Expected[i]));
+            }
+
+            R range{input};
+            const same_as<prev_permutation_result<iterator_t<R>>> auto result =
+                prev_permutation(range, ranges::less{}, get_val);
+            assert(result.in == range.end());
+            assert(!result.found);
+            assert(equal(input, Expected[number_of_permutations - 1]));
+        }
+
+        { // Validate iterator overload
+            int_wrapper input[ranges::size(Expected[0])];
+            ranges::copy(Expected[number_of_permutations - 1], input);
+
+            for (int i = number_of_permutations - 1; i-- > 0;) {
+                R range{input};
+                const same_as<prev_permutation_result<iterator_t<R>>> auto result =
+                    prev_permutation(range.begin(), range.end(), ranges::less{}, get_val);
+                assert(result.in == range.end());
+                assert(result.found);
+                assert(equal(input, Expected[i]));
+            }
+
+            R range{input};
+            const same_as<prev_permutation_result<iterator_t<R>>> auto result =
+                prev_permutation(range.begin(), range.end(), ranges::less{}, get_val);
+            assert(result.in == range.end());
+            assert(!result.found);
+            assert(equal(input, Expected[number_of_permutations - 1]));
+        }
+    }
+};
+
+struct empty_range_test {
+    template <ranges::bidirectional_range R>
+    static constexpr void call() {
+        using ranges::next_permutation, ranges::next_permutation_result, ranges::prev_permutation,
+            ranges::prev_permutation_result, ranges::equal, ranges::is_sorted, ranges::iterator_t;

```

Comment:
It looks like @StephanTLavavej's review comment re: `is_sorted` hasn't been addressed yet.

Reply:
Ouch - fixed. `ranges::equal` is also unused in this function.